### PR TITLE
Adding pypy github action 

### LINKED ISSUE #235: Add PyPy to GitHub Actions
Looks like PyPy and PyPy3 have been added to the [runners](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners). Probably just need to look at how the tests are run under tox for PyPy and then duplicate it as another job under `Test`.

### PR DESCRIPTION
**Pull request by Sangarshanan**
Closes #235 

Opening a cleaner PR with some changes suggested by @pradyunsg

### CODE REVIEW

**Comment by brettcannon** [comment]
@Sangarshanan do you mind editing a Python file to trigger the workflow so we can see how it works out? Like fix a spelling mistake in a docstring or just tweak a `.py` file slightly which is easy to revert before we merge?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+    - uses: actions/setup-python@v1
+      name: Install ${{ matrix.pypy_version }}
+      with:
+        pypy-version: ${{ matrix.pypy_version }}
+        architecture: x64
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install -r dev-requirements.txt
```
Comment (brettcannon) [inline .github/workflows/test.yml]:
```suggestion
        python -m pip install -r dev-requirements.txt
        
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        pypy_version: ['pypy2', 'pypy3']
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install ${{ matrix.pypy_version }}
+      with:
+        pypy-version: ${{ matrix.pypy_version }}
```
Comment (brettcannon) [inline .github/workflows/test.yml]: Does this work? Based on https://github.com/actions/setup-python/blob/master/action.yml it seems like it should be:
```suggestion
        python-version: ${{ matrix.pypy_version }}
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install ${{ matrix.pypy_version }}
+      with:
+        python-version: ${{ matrix.pypy_version }}
+        architecture: x64
+
+    - name: Install dependencies
+      run: |
```
Comment (brettcannon) [inline .github/workflows/test.yml]: No need to split this across a second line.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+        pypy_version: ['pypy2', 'pypy3']
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install ${{ matrix.pypy_version }}
+      with:
+        python-version: ${{ matrix.pypy_version }}
+        architecture: x64
```
Comment (brettcannon) [inline .github/workflows/test.yml]: Is this necessary? It might be better to just let the tool choose the CPU architecture (e.g. there is no x64 build of PyPy on Windows).

**Comment by Sangarshanan** [comment]
Not sure as to why there are no logs for these failures

**Comment by pradyunsg** [comment]
FYI -- there's a button on the UI for GitHub checks ("Rerun all checks") that I just pressed.

**Comment by brettcannon** [comment]
@Sangarshanan Maybe try working backwards by scaling back the pypy job run-by-run and see if that changes anything?

@pradyunsg the re-run still leads to no output unfortunately :/

**Comment by Sangarshanan** [comment]
Was an issue with the pytest faulthandler https://github.com/pytest-dev/pytest/issues/5807

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        pypy_version: ['pypy2', 'pypy3']
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - uses: actions/setup-python@v1
+      name: Install Python ${{ matrix.pypy_version }}
```
Comment (brettcannon) [inline .github/workflows/test.yml]:
```suggestion
      name: Install ${{ matrix.pypy_version }}
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -36,3 +36,32 @@ jobs:
...
+      with:
+        python-version: ${{ matrix.pypy_version }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install coverage pretend pytest==4.6.5
```
Comment (brettcannon) [inline .github/workflows/test.yml]: Can you add a comment about the problem and a link to the pytest issue so we can unpin the version once this is fixed in a released version of pytest?

Comment (Sangarshanan) [reply .github/workflows/test.yml]: Will do

File: .github/workflows/test.yml (hunk 5 above)
Comment (Sangarshanan) [reply .github/workflows/test.yml]: Oh yes 😅

**Comment by Sangarshanan** [comment]
Thanks for being awesome and helping me out @brettcannon < 3

**Comment by Sangarshanan** [comment]
Do I need to make any more changes @brettcannon  @pradyunsg ?

**Comment by brettcannon** [comment]
@Sangarshanan nope, just had to wait until one of had time to review it one last time.

Thanks for making this happen!

**Comment by aixtools** [comment]
Hi. While to update #190 I got an error from pypy testing. Not understanding how my patch was breaking things - I opened a new scratch PR that I was going to use to slowly add things to see when it broke. But I see that "master" is not passing for me - which do not understand because this PR #243 did pass all tests.

Anyway - my scratch #244 fails with https://github.com/pypa/packaging/pull/244/checks?check_run_id=339889206 Please consider reverting this addition - as I am stumped.

**Comment by brettcannon** [comment]
@aixtools (un)fortunately `master` still passes untouched with this PR so reverting it for another PR doesn't seem prudent to me.
